libs/src/sdk/api/tracks/TracksApi.ts

```
@@ -0,0 +1,128 @@
+   * Get the url of the track's streamable mp3 file
+   */
+  async streamTrack(requestParameters: StreamTrackRequest): Promise<string> {
+    if (
+      requestParameters.trackId === null ||
+      requestParameters.trackId === undefined
+    ) {
+      throw new RequiredError(
+        'trackId',
+        'Required parameter requestParameters.trackId was null or undefined when calling getTrack.'
+      )
+    }
+
+    const path = `/tracks/{track_id}/stream`.replace(
+      `{${'track_id'}}`,
+      encodeURIComponent(String(requestParameters.trackId))
+    )
+    const host = await this.discoveryNodeSelectorService.getSelectedEndpoint()
+    return `${host}${BASE_PATH}${path}`
+  }
+
+  /**
+   * Upload a track
+   */
+  async uploadTrack(requestParameters: UploadTrackRequest) {
+    const { trackFile, coverArtFile, metadata, onProgress } = requestParameters
+
+    // Validate inputs
+
+    if (!isFileValid(trackFile)) {
+      throw new Error('Track file is not valid')
+    }
+
+    if (!isFileValid(coverArtFile)) {
+      throw new Error('Cover art file is not valid')
+    }
+
+    if (typeof metadata !== 'object') {
+      throw new Error('Metadata object is not valid')
+    }
+
+    requestParameters.metadata.owner_id = Number(requestParameters.artistId)
+
+    const metadataMissingProperties = objectMissingProperties(
+      requestParameters.metadata,
+      TRACK_REQUIRED_PROPERTIES
+    )
+    if (metadataMissingProperties) {
+      throw new Error(
+        `Metadata object is missing properties: ${metadataMissingProperties}`
+      )
+    }
+
+    const metadataMissingValues = objectMissingProperties(
+      requestParameters.metadata,
+      TRACK_REQUIRED_VALUES
+    )
+    if (metadataMissingValues) {
+      throw new Error(
+        `Metadata object is missing values: ${metadataMissingValues}`
+      )
+    }
+
+    // Upload track audio and cover art to storage node
+    const updatedMetadata = await this.storage.uploadTrackAudioAndCoverArt(
```

should we wrap this in try/catch and stick the error in the return value below (if it throws)?

---

No for consistency with the rest of the generated sdk this should throw an error

---

icic